### PR TITLE
gtls: Values stored to but never read

### PR DESCRIPTION
**Pull request by bagder**
Detected by clang-tidy

### CODE REVIEW

File: lib/vtls/gtls.c [hunk 1]
```diff
@@ -1679,7 +1678,6 @@ static int Curl_gtls_shutdown(struct connectdata *conn, int sockindex)
       else if(0 == what) {
         /* timeout */
         failf(data, "SSL shutdown timeout");
-        done = 1;
```
Comment (danielgustafsson) [inline lib/vtls/gtls.c]: Removing the `break` statement should have the same effect, but be a little more consistent with how the loop runs shouldn't it? Ie, the `else` case doens't `break` but sets `done = 1`

Comment (bagder) [reply lib/vtls/gtls.c]: Ack. That's nicer...

Comment (bagder) [reply lib/vtls/gtls.c]: I'm taking the opportunity and change the variable type to a bool as well...

Comment (danielgustafsson) [reply lib/vtls/gtls.c]: +1, that's a good idea. Maybe we can slip this hunk into the patch while we're at it?
```
diff --git a/lib/vtls/openssl.c b/lib/vtls/openssl.c
index 4d1a95ed4..b06705312 100644
--- a/lib/vtls/openssl.c
+++ b/lib/vtls/openssl.c
@@ -1262,7 +1262,7 @@ static int Curl_ossl_shutdown(struct connectdata *conn, int sockindex)
   ssize_t nread;
   int buffsize;
   int err;
-  int done = 0;
+  bool done = FALSE;
 
   /* This has only been tested on the proftpd server, and the mod_tls code
      sends a close notify alert without waiting for a close notify alert in
@@ -1290,7 +1290,7 @@ static int Curl_ossl_shutdown(struct connectdata *conn, int sockindex)
         case SSL_ERROR_ZERO_RETURN: /* no more data */
           /* This is the expected response. There was no data but only
              the close notify alert */
-          done = 1;
+          done = TRUE;
           break;
         case SSL_ERROR_WANT_READ:
           /* there's data pending, re-invoke SSL_read() */
@@ -1299,7 +1299,7 @@ static int Curl_ossl_shutdown(struct connectdata *conn, int sockindex)
         case SSL_ERROR_WANT_WRITE:
           /* SSL wants a write. Really odd. Let's bail out. */
           infof(data, "SSL_ERROR_WANT_WRITE\n");
-          done = 1;
+          done = TRUE;
           break;
         default:
           /* openssl/ssl.h says "look at error stack/return value/errno" */
@@ -1309,20 +1309,20 @@ static int Curl_ossl_shutdown(struct connectdata *conn, int sockindex)
                  ossl_strerror(sslerror, buf, sizeof(buf)) :
                  SSL_ERROR_to_str(err)),
                 SOCKERRNO);
-          done = 1;
+          done = TRUE;
           break;
         }
       }
       else if(0 == what) {
         /* timeout */
         failf(data, "SSL shutdown timeout");
-        done = 1;
+        done = TRUE;
       }
       else {
         /* anything that gets here is fatally bad */
         failf(data, "select/poll on SSL socket, errno: %d", SOCKERRNO);
         retval = -1;
-        done = 1;
+        done = TRUE;
       }
     } /* while()-loop for the select() */
```

Comment (bagder) [reply lib/vtls/gtls.c]: done!

**Pull request closed without merging**